libs/wingc/src/lib.rs

```diff
@@ -254,6 +257,91 @@ fn add_builtin(name: &str, typ: Type, scope: &mut Scope, types: &mut Types) {
 		.expect("Failed to add builtin");
 }
 
+/// The different phases of compilation, used for tracking compilation context
```

Comment:
(style) WDYT of putting these types and macros into a dedicated file/module?

Reply:
Yes, good idea. I'll do it.